imports/client/components/PuzzleListPage.tsx

```diff
@@ -424,37 +428,52 @@ interface PuzzleListPageProps extends PuzzleListPageWithRouterParams {
   canUpdate: boolean;
   allPuzzles: PuzzleType[];
   allTags: TagType[];
+  hunt?: HuntType;
```

Comment:
Make this just `hunt: HuntType;` and use a TS assertion `!` below where you put it in the object.

(separately: in most places I'd rather include keys in objects and have their values possibly be `undefined` than have optional keys, since that's generally how we get data out of Meteor so it's less noisy and I anticipate it being slightly easier for the JIT to optimize, since there's only one set of fields to recognize for the "struct".  The way to express that would be `hunt: HuntType | undefined;`)

Reply:
Summarizing a productive Slack conversation with @zarvox:

Since HuntApp already takes care of the hunt subscription and catches non-existent hunts, the assertion is reasonable in this case and, from a maintenance perspective, has essentially the same impact as removing the redundant subscription. In the more general case, our preference would be to use required props that may be undefined rather than optional props. Either approach eliminates awkward constructions, like the conditional spread below, that are required with optional props.

Optional props are used for this case in quite a few other places, which we may want to clean up.